dotnet-client-libraries/HowToUseSftpForSecureFileUploadInDotNet/Test/SshNetSftpClientLiveTests.cs

```diff
@@ -0,0 +1,123 @@
+{
+    private readonly ITestOutputHelper _output;
+
+    public SshNetSftpClientLiveTests(ITestOutputHelper output)
+    {
+        _output = output;
+    }
+
+    [Fact]
+    public void WhenUsingSshNet_ThenCanConnectUsingUsernameAndPassword()
+    {
+        using SftpClient client = new("192.168.1.42", 2222, "tester", "password");
+        client.Connect();
+
+        Assert.True(client.IsConnected);
+
+        client.Disconnect();
+    }
+
+    [Fact]
+    public void WhenUsingSshNet_ThenCanConnectUsingPrivateKey()
+    {
+        var privateKey = new PrivateKeyFile("./ssh-keys/userkey");
+        using SftpClient client = new("192.168.1.42", 2222, "tester", privateKey);
+        client.Connect();
+
+        Assert.True(client.IsConnected);
+
+        client.Disconnect();
+    }
+
+    [Fact]
+    public void WhenUsingSshNet_ThenCanCheckHostFingerprint()
+    {
+
+        var privateKey = new PrivateKeyFile("./ssh-keys/userkey");
+
+        // Change the hardcoded fingerprint to match your test server's.
+        var expectedFingerprint = Convert.FromHexString("D29B9B75F725D9164EA0D4935AFCACD3");
+
+        using SftpClient client = new("192.168.1.42", 2222, "tester", privateKey);
+        client.HostKeyReceived += delegate (object? sender, HostKeyEventArgs e)
+        {
+            e.CanTrust = e.FingerPrint.SequenceEqual(expectedFingerprint);
+        };
+
+        client.Connect();
+
+        Assert.True(client.IsConnected);
+
+        client.Disconnect();
+    }
+
+    [Fact]
+    public void WhenHostFingerprintCheckFails_ThenTheExceptionsIsThrown()
+    {
+
+        var privateKey = new PrivateKeyFile("./ssh-keys/userkey");
+        var expectedFingerprint = Convert.FromHexString("D29B9B75F725D9164EA0D4935AFCACBB"); // Wrong host fingerprint.
+
+        using SftpClient client = new("192.168.1.42", 2222, "tester", privateKey);
+        client.HostKeyReceived += delegate (object? sender, HostKeyEventArgs e)
+        {
+            e.CanTrust = e.FingerPrint.SequenceEqual(expectedFingerprint);
+        };
+
+        Assert.Throws<SshConnectionException>(() => client.Connect());
+    }
+
+    [Fact]
+    public void WhenUsingSshNet_ThenCanUploadFile()
+    {
+        using SftpClient client = new("192.168.1.42", 2222, "tester", "password");
+
+        try
+        {
+            client.Connect();
+            if (client.IsConnected)
+            {
+                client.CreateDirectory("/images");
+                client.UploadFile(File.OpenRead("./files/portrait.jpg"), "/images/portrait.jpg");
+
+                _output.WriteLine("Directory listing:");
+                foreach(var sftpFile in client.ListDirectory("/images"))
+                {
+                    _output.WriteLine($"\t{sftpFile.FullName}");
+                }
+
+                client.Disconnect();
+            }
+        }
+        catch (Exception e)
+        when (e is SshConnectionException || e is SocketException || e is ProxyException)
```

Comment:
Here, we can switch to using patterns : 
```suggestion
                    when (e is SshConnectionException or SocketException or ProxyException)
```